### ISSUE TITLE
Misleading error message when calling method on unowned struct

### ISSUE THREAD

**Issue by frewsxcv** [rust-lang/rust#24582]
#### Code

```rust
fn main() {
    let x: Vec<u16> = vec![1, 2, 3];
    let x_ref: &Vec<u16> = &x;
    let x_deref: Vec<u16> = (*x_ref);
    x_deref.map(|z| z + 1);
}
```

#### Compiler error

```
<anon>:5:13: 5:27 error: type `collections::vec::Vec<u16>` does not implement any method in scope named `map`
<anon>:5     x_deref.map(|z| z + 1);
                     ^~~~~~~~~~~~~~
error: aborting due to previous error
playpen: application terminated with error code 101
```

I personally understand what the issue is, but to claim that `Vec` doesn't implement `map` is incorrect.

#### Playpen

http://is.gd/Auf2Af

**Comment by Stebalien** [comment]
Vec doesn't implement map.

**Comment by frewsxcv** [comment]
This is what happens when I open Issues before I consume caffeine. Sorry for the noise.
